### DISCUSSION TITLE
login with different password column name in database

### DISCUSSION THREAD

**Discussion by moamahfouz** [laravel/framework#44562] [Q&A] [answered]
for some reason I had to deal with an old database table that contain users table. Password column is named as `password_hash`.

How can I login by using
```php
Auth::attempt([....]) 
```

When it comes to login. I already know that we can override email with 

```php  
public function username(){
   return 'username';
}
```

Is there any similar function to password?
Thanks.

**Comment by erikn69** [comment, marked as answer]
On user
https://github.com/laravel/framework/blob/b9203fca96960ef9cd8860cb4ec99d1279353a8d/src/Illuminate/Auth/Authenticatable.php#L49-L52

**Reply by blockhunts** [reply]
To explain erikn69's answer:

In default configuration, go to _app/Models/user.php_

add this code inside the user class

```
public function getAuthPassword()
    {
        return $this->{your password column name};
    }
```

**Comment by arbazargani** [comment]
in app/Models/user.php
add following method:
```
public function getAuthPassword() 
 { 
     return $this->password_hash; 
 }
```
